customdomain-terraform/modules/customdomain/main.tf

```diff
@@ -0,0 +1,91 @@
+variable "zone_id" {
+  type = "string"
+}
+
+variable "govuk_domain" {
+  type = "string"
+}
+
+variable "govsvcuk_domain" {
+  type = "string"
+}
+
+variable "aws_account_role_arn" {
+  type = "string"
+}
+
+provider "aws" {
+  alias  = "us-east-1"
+  region = "us-east-1"
+
+  assume_role {
+    role_arn = var.aws_account_role_arn
+  }
+}
+
+resource "aws_acm_certificate" "cert" {
+  provider          = aws.us-east-1
+  domain_name       = var.govuk_domain
+  validation_method = "DNS"
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_cloudfront_distribution" "cloudfront_dist" {
+  enabled          = true
+  is_ipv6_enabled  = true
+  price_class      = "PriceClass_100" # EU and NA optimised, lower cost
+  aliases          = [var.govuk_domain]
+
+  viewer_certificate {
+    acm_certificate_arn = aws_acm_certificate.cert.arn
```

Comment:
I don't see anything doing the validation dance? is that intentional? ... I thought I would see a `aws_acm_certificate_validation` resource in here no?

Oh... is that just because you know it would fail to validate until you have the zone delegated maybe

Reply:
yeah we probably need to come back and add that when we uncomment references to this module